controllers/helper_test.go

```diff
@@ -164,6 +166,37 @@ func testBuildInMemoryCredentialsSecret(name, ns string) *corev1.Secret {
 	}
 }
 
+// testEndpointsTraversable consumes an array of endpoints and returns a boolean
+// indicating presence of that path from host to all destinations
+// this function DOES NOT report a presence of an endpoint with one of destinations DNSNames
+func testEndpointsTraversable(endpoints []*endpoint.Endpoint, host string, destinations []string) bool {
```

Comment:
it will create confusion since we have `endpoint.targets` and a list of `targets` we are trying to reach. They were called `targets` and `endpointTargets` for a while but this does not help. But yes, i really want to avoid calling them `targets` as much as I can: the function is awkward to read as is and creating similar names won't help.